### PR TITLE
Add comprehensive test suite for Languages controller

### PR DESCRIPTION
**Pull request by ravidulundu**
Screenshot from 2025-06-13 00-45-37
What's Changed

✅ Complete test coverage for all CRUD operations (findAll, findOne, create, update, remove) ✅ Error handling tests for edge cases (404, 409, 500 responses) ✅ Database mocking with chainable query builder simulation ✅ Brute-force fuzzing with 1,000 random payloads to stress-test controller ✅ BDD-style naming for better test readability

Key Improvements

Mocked dependencies: Clean separation with mock.module() for database layer
Helper functions: Reusable makeLang() factory and ctx() mock creator
Edge case coverage: Duplicate codes, missing records, DB failures
Stress testing: Brute-force fuzzing to catch unexpected edge cases and ensure stability

Testing Framework

Bun Test with TypeScript support
Hono Context mocking for HTTP layer
Drizzle ORM query simulation

All tests pass ✨

## Summary by CodeRabbit

- **Chores**
  - Updated ignore rules to exclude Drizzle migration folders while preserving specific journal files.
  - Removed example environment variable file from the web app.
  - Added a test script for the API and a test task to the build system.

- **Refactor**
  - Migrated database schema from an event-centric to a game-centric model, introducing new tables and relationships for games, platforms, and user profiles.

- **Bug Fixes**
  - Corrected the sidebar navigation link for Languages in the dashboard.

- **Tests**
  - Replaced the language controller test suite with a new, comprehensive test suite covering all language-related API endpoints, including edge cases and fuzz testing.

- **Style**
  - Updated import paths in the dashboard languages page for consistency.
  - Improved language flag rendering using optimized image components and refined UI text encoding.

### CODE REVIEW

File: apps/api/drizzle/meta/_journal.json [hunk 1]
```diff
@@ -5,8 +5,8 @@
     {
       "idx": 0,
       "version": "7",
-      "when": 1749627422474,
-      "tag": "0000_reflective_wrecker",
+      "when": 1749773175806,
+      "tag": "0000_regular_titania",
```
[the remark this answers is not in the document]
Comment (ravidulundu) [reply apps/api/drizzle/meta/_journal.json:9]: I forgot to add this friend to .gitignore.
